QuadRelate.Helpers/BoardHelper.cs

```diff
@@ -24,20 +24,47 @@ private static bool DoesHorizontalWinExist(this Board board)
         {
             for (var y = 0; y < Board.Height; y++)
             {
+                var inARowCount = 0;
```

Comment:
This is still a bit complicated.
I think it can be done in two nested for loops.
If you are happy that the unit tests are correct - you can refactor and try to simplify.